zebrad/src/commands/start.rs

```diff
@@ -154,25 +153,91 @@ impl StartCmd {
+        // ongoing futures & tasks
+        pin!(syncer_error_future);
+        pin!(mempool_crawler_task_handle);
+        pin!(mempool_queue_checker_task_handle);
+        pin!(tx_gossip_task_handle);
 
-            mempool_queue_result = mempool_queue_checker_task_handle.fuse() => mempool_queue_result
-                .expect("unexpected panic in the mempool queue checker")
-                .map_err(|e| eyre!(e)),
+        // startup tasks
+        let groth16_download_handle_fused = (&mut groth16_download_handle).fuse();
+        pin!(groth16_download_handle_fused);
 
-            tx_gossip_result = tx_gossip_task_handle.fuse() => tx_gossip_result
-                .expect("unexpected panic in the transaction gossip task")
-                .map_err(|e| eyre!(e)),
-        }
+        // Wait for tasks to finish
+        let exit_status = loop {
+            let mut exit_when_task_finishes = true;
+
+            let result = select! {
+                // We don't spawn the syncer future into a separate task yet.
+                // So syncer panics automatically propagate to the main zebrad task.
+                sync_result = &mut syncer_error_future => sync_result
+                    .map(|_| info!("syncer task exited")),
+
+                sync_gossip_result = &mut sync_gossip_task_handle => sync_gossip_result
+                    .expect("unexpected panic in the chain tip block gossip task")
+                    .map(|_| info!("chain tip block gossip task exited"))
+                    .map_err(|e| eyre!(e)),
+
+                mempool_crawl_result = &mut mempool_crawler_task_handle => mempool_crawl_result
+                    .expect("unexpected panic in the mempool crawler")
+                    .map(|_| info!("mempool crawler task exited"))
+                    .map_err(|e| eyre!(e)),
+
+                mempool_queue_result = &mut mempool_queue_checker_task_handle => mempool_queue_result
+                    .expect("unexpected panic in the mempool queue checker")
+                    .map(|_| info!("mempool queue checker task exited"))
+                    .map_err(|e| eyre!(e)),
+
+                tx_gossip_result = &mut tx_gossip_task_handle => tx_gossip_result
+                    .expect("unexpected panic in the transaction gossip task")
+                    .map(|_| info!("transaction gossip task exited"))
+                    .map_err(|e| eyre!(e)),
+
+                // Unlike other tasks, we expect the download task to finish while Zebra is running.
+                groth16_download_result = &mut groth16_download_handle_fused => {
+                    groth16_download_result
+                        .unwrap_or_else(|_| panic!(
+                            "unexpected panic in the Groth16 pre-download and check task. {}",
+                            zebra_consensus::groth16::Groth16Params::failure_hint())
+                        );
+
+                    info!("Groth16 pre-download and check task finished");
```

Comment:
Nit: Same as other `pre-download` suggestions:

```suggestion
                            "unexpected panic in the Groth16 download and check task. {}",
                            zebra_consensus::groth16::Groth16Params::failure_hint())
                        );

                    info!("Groth16 download and check task finished");
```